src/main/java/uk/gov/hmcts/ccd/domain/service/getcase/GetCaseCallback.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.hmcts.ccd.domain.service.getcase;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CaseViewField;
+import uk.gov.hmcts.ccd.domain.model.callbacks.GetCaseCallbackResponse;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+import uk.gov.hmcts.ccd.domain.service.stdapi.CallbackInvoker;
+import uk.gov.hmcts.ccd.endpoint.exceptions.CallbackException;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+public class GetCaseCallback {
+    private static final Logger LOG = LoggerFactory.getLogger(GetCaseCallback.class);
+
+    private final CallbackInvoker callbackInvoker;
+
+    public GetCaseCallback(CallbackInvoker callbackInvoker) {
+        this.callbackInvoker = callbackInvoker;
+    }
+
+    public GetCaseCallbackResponse invoke(final CaseTypeDefinition caseTypeDefinition, final CaseDetails caseDetails,
+                                          final List<CaseViewField> metadataFields) {
+        GetCaseCallbackResponse response;
+        try {
+            ResponseEntity<GetCaseCallbackResponse> getCaseCallbackResponse = callbackInvoker
+                .invokeGetCaseCallback(caseTypeDefinition, caseDetails);
+
+            response = getCaseCallbackResponse.getBody();
+        } catch (Exception e) {
+            LOG.error("CCD_CDI_CallbackGetCaseUrl: " + e.getMessage());
+            throw new CallbackException(e.getMessage());
+        }
+
+        validateNewMetadataFields(metadataFields, response);
+
```

Comment:
If the expectation is that the response is null you could add Optional.

Reply:
Same as above